### PR TITLE
Bring your own workers

### PR DESCRIPTION
**Pull request by gcoonrod**
Extending/modifying the current functionality of Resque within ActionHero is currently very cumbersome and has made upgrading our deployments very troublesome. I am proposing adding a configuration section to `config/tasks.js` to allow developers to replace the existing queues, scheduler, and multiWorkers with custom implementations. 

An example extension to queues: 
```javascript
'use strict';

var NR = require('node-resque');
var pluginRunner = require('../node_modules/node-resque/lib/pluginRunner.js');

//noinspection JSUnresolvedVariable
module.exports = {
  loadPriority:  590,
  initialize: function(api, next){
    api.customResque = {};
    
    let customQueue = NR.queue;
    
    customQueue.prototype.enqueueFront = function(q, func, args, callback){
      var self = this;
      if(arguments.length === 3 && typeof args === 'function'){
        callback = args;
        args = [];
      }else if(arguments.length < 3){
        args = [];
      }

      args = arrayify(args);
      var job = self.jobs[func];
      pluginRunner.runPlugins(self, 'before_enqueue', func, q, job, args, function(err, toRun){
        if(toRun === false){
          if(typeof callback === 'function'){ callback(err, toRun); }
        }else{
          self.connection.redis.sadd(self.connection.key('queues'), q, function(){
            self.connection.redis.lpush(self.connection.key('queue', q), self.encode(q, func, args), function(){
              pluginRunner.runPlugins(self, 'after_enqueue', func, q, job, args, function(){
                if(typeof callback === 'function'){ callback(err, toRun); }
              });
            });
          });
        }
      });
    };
    
    api.customResque.queue = customQueue;

    next();
  }
};
```

### CODE REVIEW

**Comment by gcoonrod** [comment]
Looking for comments before making final changes. This current version will fail tests.

File: config/tasks.js [hunk 1]
```diff
@@ -43,6 +43,12 @@ exports['default'] = {
       toDisconnectProcessors: true,
       // What redis server should we connect to for tasks / delayed jobs?
       redis: api.config.redis,
+      // Customize Resque primitives, replace null with required replacement.
+      resque: {
```
Comment (evantahler) [inline config/tasks.js]: I would call this something like `resque_overrides` or similar, to help indicate the function

**Comment by evantahler** [comment]
one note, otherwise looks good!

**Comment by evantahler** [comment]
ready to merge?

**Comment by gcoonrod** [comment]
I'm ready.

**Comment by evantahler** [comment]
Please add your example (or something like it) to the docs, and we'll be ready to go! 

https://github.com/evantahler/actionhero/blob/master/site/source/includes/docs/core/tasks.md

**Comment by gcoonrod** [comment]
Docs updated. Let me know if I need to add more detail.

**Comment by evantahler** [comment]
👍

**Pull request merged**